packages/astro/src/vite-plugin-jsx/index.ts

```diff
@@ -137,9 +137,31 @@ export default function jsx({ config, logging }: AstroPluginJSXOptions): Plugin
 				JSX_RENDERER_CACHE.set(config, jsxRenderers);
 			}
 
-			// Attempt: Single JSX renderer
+			const astroRenderer = jsxRenderers.get('astro');
+
+			// Shortcut: only use Astro renderer for MD and MDX files
+			if ((id.includes('.mdx') || id.includes('.md')) && astroRenderer) {
```

Comment:
Would leave this to @natemoo-re but I'm wondering if the astro renderer is intended to be used outside of of just MDX. Otherwise why would it be in astro core and not in the mdx package?

Reply:
It might be nice to have the jsx renderer API allow you to associate a file extension with a jsx renderer. That way mdx can say that it handles .mdx files while leaving tsx/jsx untouched.

Reply:
The Astro JSX renderer certainly _could_ be used for `.[jt]sx` files in the future, but Ben and I discussed this approach as temporary perf shortcut that avoids import scanning for `.mdx` and `.md` files since we already know what runtime they should use.

I like the idea of opening up the JSX renderer API to support explicit file extensions in the future! Sounds like a great v1.1.0 enhancement.